index.js

```diff
@@ -197,26 +214,90 @@ function uploadMergeFiles(viewId, mergeData, templatesToRecords){
+    if ( promises.length > 0 &&
+        table == "merged" &&
+        !findDocFieldName(view[0].structure, config.merged_doc_table.merged_doc_details_field_name) &&
+        findDocFieldName(view[0].structure, config.merged_doc_table.merged_doc_to_template_relationship_field_name) &&
+        findDocFieldName(view[0].structure, config.merged_doc_table.merged_document_field_name) &&
+        findDocFieldName(view[0].structure, config.merged_doc_table.merge_user_field_name) ) {
+      return log.error(`Unable to find details field from ${table} table, please ensure EXACT match`);
+    } else if ( promises.length > 0 &&
+        table == "merged" &&
+        findDocFieldName(view[0].structure, config.merged_doc_table.merged_doc_details_field_name) &&
+        !findDocFieldName(view[0].structure, config.merged_doc_table.merged_doc_to_template_relationship_field_name) &&
+        findDocFieldName(view[0].structure, config.merged_doc_table.merged_document_field_name) &&
+        findDocFieldName(view[0].structure, config.merged_doc_table.merge_user_field_name) ) {
+      return log.error(`Unable to find template relationship from ${table} table, please ensure EXACT match`);
+    } else if ( promises.length > 0 &&
+        table == "merged" &&
+        findDocFieldName(view[0].structure, config.merged_doc_table.merged_doc_details_field_name) &&
+        findDocFieldName(view[0].structure, config.merged_doc_table.merged_doc_to_template_relationship_field_name) &&
+        !findDocFieldName(view[0].structure, config.merged_doc_table.merged_document_field_name) &&
+        findDocFieldName(view[0].structure, config.merged_doc_table.merge_user_field_name) ) {
+      return log.error(`Unable to find field name from ${table} table, please ensure EXACT match`);
+    } else if ( promises.length > 0 &&
+        table == "merged" &&
+        findDocFieldName(view[0].structure, config.merged_doc_table.merged_doc_details_field_name) &&
+        findDocFieldName(view[0].structure, config.merged_doc_table.merged_doc_to_template_relationship_field_name) &&
+        findDocFieldName(view[0].structure, config.merged_doc_table.merged_document_field_name) &&
+        !findDocFieldName(view[0].structure, config.merged_doc_table.merge_user_field_name) ) {
+      return log.error(`Unable to find user field name from ${table} table, please ensure EXACT match`);
+    } else if ( promises.length > 0 &&
+        table == "template" &&
+        !findDocFieldName(view[0].structure, config.template_table.field_name_for_template_document) ) {
+      return log.error(`Unable to find field name from ${table} table, please ensure EXACT match`);
+    } else {
+      return log.error(`Could not find ${table} view, please check the view id`);
+    }
+  })
+}
+
+/**
+ * Finds the merged document field name from an array of table columns
+ * @param {Array} fields
+ */
+function findDocFieldName(fields, findName) {
+  var found = false;
+  for (var i = 0; i < fields.length; i++) {
```

Comment:
So the issue here is in a worse case you'll hit this code 5 times. so `O(5n)` which isn't terrible especially since if our config file ever grew to a size where that started to matter we have way bigger problems then a for loop, but there's no since in letting bad habits persist.

I'd suggest for turning the structure from the API into a map/object (all the same in JS) keyed by name, `O(n)` then every access to determine if a value exists becomes `O(1)`. Something like:
```JavaScript
var fieldNames = {};
fields.forEach((field) => {fieldNames[field.name] = true});

//then you can just say
if(!fieldNames[questionableField]){
    showError();
}
```